lib/functions.php

```diff
@@ -762,6 +762,31 @@ function concat(array $iterators): Iterator
         return $emitter->iterate();
     }
 
+    /**
+     * Discards all remaining items and returns the number of discarded items.
+     *
+     * @template TValue
+     *
+     * @param Iterator $iterator
+     *
+     * @return Promise
+     *
+     * @psalm-param Iterator<TValue> $iterator
+     * @psalm-return Promise<int>
+     */
+    function discard(Iterator $iterator): Promise
```

Comment:
I'd name it `count`, but I guess it might be misused and people will complain that the iterator can't be reused then.

Reply:
@trowski Should it return `Promise<void>` instead of should we keep the count?

Reply:
Keep the count.